image/internal/manifest/manifest_test.go

```diff
@@ -73,6 +73,65 @@ func TestDigest(t *testing.T) {
 	assert.Equal(t, digest.Digest(digestSha256EmptyTar), actualDigest)
 }
 
+func TestDigestWithAlgorithm(t *testing.T) {
+	sha256Cases := []struct {
+		path           string
+		expectedDigest digest.Digest
+	}{
+		{"v2s2.manifest.json", TestDockerV2S2ManifestDigest},
+		{"v2s1.manifest.json", TestDockerV2S1ManifestDigest},
+		{"v2s1-unsigned.manifest.json", TestDockerV2S1UnsignedManifestDigest},
+	}
+	for _, c := range sha256Cases {
+		manifest, err := os.ReadFile(filepath.Join("testdata", c.path))
+		require.NoError(t, err)
+		actualDigest, err := DigestWithAlgorithm(manifest, digest.SHA256)
+		require.NoError(t, err)
+		assert.Equal(t, c.expectedDigest, actualDigest, c.path)
+		defaultDigest, err := Digest(manifest)
+		require.NoError(t, err)
+		assert.Equal(t, defaultDigest, actualDigest, c.path)
+	}
+
+	// Test with SHA512
+	for _, c := range sha256Cases {
+		manifest, err := os.ReadFile(filepath.Join("testdata", c.path))
+		require.NoError(t, err)
+		actualDigest, err := DigestWithAlgorithm(manifest, digest.SHA512)
+		require.NoError(t, err)
+		assert.Equal(t, digest.SHA512, actualDigest.Algorithm())
```

Comment:
This is not particularly precise…

I think
- The empty input can be reasonably included in the test case table
- The test case table can contain exact sha256 and sha512 digests

Then `TestDigest` and `TestDigestWithAlgorithm` can work from a shared table, differing only in whether they process 1 / 2 values.

Alternatively, if `Digest` became just a wrapper over `DigestWithAlgorithm`, it would be acceptable to have comprehensive tests only for `…WithAlgorithm` (more precise than the current ones), and a ~smoke-test for `Digest`. (I weakly prefer the shared table and thorough testing of both.)